src/starknet_storage/storage.rs

```diff
@@ -0,0 +1,98 @@
+use super::errors::storage_errors::StorageError;
+use cairo_rs::types::instruction::Res;
+use std::str;
+
+//* ------------------
+//*   Storage Trait
+//* ------------------
+
+/* -----------------------------------------------------------------------------------
+   -----------------------------------------------------------------------------------
+
+    This module implements the trait for the the storages operations.
+    The trait default functions handle the different data types that can be stored,
+    some assumptions and work arounds have been taken for this.
+
+    * All data types are turned into Vec<u8> before being passed to the set_value function.
+    This is due to rust restrictions on parameter types for the functions in traits.
+
+    * The same is true for get_value but this returns an Option because there may be nothing in the storage for a given key.
+
+    * float types are assumed to be f32, in case of needing to store f64, get_double and set_double functions can be implemented.
```

Comment:
Maybe we can do the opposite and default to `f64`. `f32` is just a space optimization in modern hardware.

Reply:
ok, i though it was unnecessary to use f64 but i will fix it